app/src/main/java/com/brandon/forzenbook/viewmodels/LoginViewModel.kt

```diff
@@ -0,0 +1,239 @@
+                                firstName = firstName,
+                                lastName = lastName,
+                                location = location,
+                                dateOfBirth = date,
+                                isDuplicateUser = true,
+                            )
+                        }
+                        CreateUserOutcome.CREATE_USER_SUCCESS -> {
+                            CreateAccountUiState.Loaded
+                        }
+                    }
+                }
+            } else {
+                createAccountState.value = CreateAccountUiState.Idle(
+                    isEmailError = validMail,
+                    isLocationError = validLoc,
+                    isDateError = validDob,
+                    email = email,
+                    firstName = firstName,
+                    lastName = lastName,
+                    location = location,
+                    dateOfBirth = date,
+                )
+            }
+        }
+        Log.e(VIEWMODEL_ERROR_TAG, "${createAccountState.value}")
+    }
+
+    private fun isValidLocation(location: String): Boolean {
+        return location.length <= LOCATION_CHAR_LIMIT && location.isNotEmpty()
+    }
+
+    private fun isValidEmail(email: String): Boolean {
+        return PatternsCompat.EMAIL_ADDRESS.matcher(email).matches()
+    }
+
+    private fun isValidDateOfBirth(dob: String): Boolean {
+        val date = dob.filter { it != '/' && it != '-' }
+        date.forEach {
+            if (!it.isDigit()) {
+                return false
+            }
+        }
+        return try {
+            val inputFormat = SimpleDateFormat(INCOMING_DATE_FORMAT)
+            val tempDate = inputFormat.parse(date)
+            val outputFormat = SimpleDateFormat(OUTGOING_DATE_FORMAT)
+            val dateString = tempDate?.let { outputFormat.format(it) }
+            sqlDate = Date.valueOf(dateString)
+            true
+        } catch (e: Exception) {
+            Log.e(VIEWMODEL_ERROR_TAG, "Invalid Date of Birth Provided.")
+            false
+        }
+    }
+
+    private fun isValidCode(code: String): Boolean {
+        return code.isDigitsOnly() && code.length == CODE_CHAR_LIMIT
+    }
+
+    fun isNetworkAvailable(context: Context): Boolean {
+        val connectivityManager =
+            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val activeNetworkInfo = connectivityManager.activeNetworkInfo
+        return activeNetworkInfo?.isConnected ?: false
+    }
+
+    companion object {
+        const val VIEWMODEL_ERROR_TAG = "Brandon_Test_ViewModel"
+        const val EMAIL_CHAR_LIMIT = 20
+        const val CODE_CHAR_LIMIT = 6
+        const val DATE_OF_BIRTH_CHAR_LIMIT = 8
+        const val LOCATION_CHAR_LIMIT = 64
+        const val INCOMING_DATE_FORMAT = "MMddyyyy"
+        const val OUTGOING_DATE_FORMAT = "yyyy-MM-dd"
+    }
+}
+
+enum class CreateUserOutcome {
```

Comment:
This seems so lonely out here. Maybe in a CreateUserVM?

Reply:
I'll make a new viewmodel for this, since the more of think of it, it is a little difficult to keep two states in one class